x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/time_shift.tsx

```diff
@@ -0,0 +1,390 @@
+    if (column) {
+      dateHistogramInterval = search.aggs.parseInterval(
+        search.aggs.getDateHistogramMetaDataByDatatableColumn(column)?.interval || ''
+      );
+    }
+  }
+
+  function isValueTooSmall(parsedValue: ReturnType<typeof parseTimeShift>) {
+    return (
+      dateHistogramInterval &&
+      parsedValue &&
+      typeof parsedValue === 'object' &&
+      parsedValue.asMilliseconds() < dateHistogramInterval.asMilliseconds()
+    );
+  }
+
+  function isValueNotMultiple(parsedValue: ReturnType<typeof parseTimeShift>) {
+    return (
+      dateHistogramInterval &&
+      parsedValue &&
+      typeof parsedValue === 'object' &&
+      !Number.isInteger(parsedValue.asMilliseconds() / dateHistogramInterval.asMilliseconds())
+    );
+  }
+
+  const parsedLocalValue = localValue && parseTimeShift(localValue);
+  const isLocalValueInvalid = Boolean(parsedLocalValue === 'invalid');
+  const localValueTooSmall = parsedLocalValue && isValueTooSmall(parsedLocalValue);
+  const localValueNotMultiple = parsedLocalValue && isValueNotMultiple(parsedLocalValue);
+
+  function getSelectedOption() {
+    if (!localValue) return [];
+    const goodPick = timeShiftOptions.filter(({ value }) => value === localValue);
+    if (goodPick.length > 0) return goodPick;
+    return [
+      {
+        value: localValue,
+        label: localValue,
+      },
+    ];
+  }
+
+  return (
+    <div
+      ref={(r) => {
+        if (r && isFocused) {
+          const timeShiftInput = r.querySelector('[data-test-subj="comboBoxSearchInput"]');
+          if (timeShiftInput instanceof HTMLInputElement) {
+            timeShiftInput.focus();
+          }
+        }
+      }}
+    >
+      <EuiFormRow
+        display="columnCompressed"
+        fullWidth
+        label={i18n.translate('xpack.lens.indexPattern.timeShift.label', {
+          defaultMessage: 'Time shift',
+        })}
+        helpText={i18n.translate('xpack.lens.indexPattern.timeShift.help', {
+          defaultMessage: 'Enter the time shift number and unit',
+        })}
+        error={
+          (localValueTooSmall &&
+            i18n.translate('xpack.lens.indexPattern.timeShift.tooSmallHelp', {
+              defaultMessage:
+                'Time shift should to be larger than the date histogram interval. Either increase time shift or specify smaller interval in date histogram',
+            })) ||
+          (localValueNotMultiple &&
+            i18n.translate('xpack.lens.indexPattern.timeShift.noMultipleHelp', {
+              defaultMessage:
+                'Time shift should be a multiple of the date histogram interval. Either adjust time shift or date histogram interval',
+            }))
+        }
+        isInvalid={Boolean(isLocalValueInvalid || localValueTooSmall || localValueNotMultiple)}
+      >
+        <EuiFlexGroup gutterSize="s" alignItems="center">
+          <EuiFlexItem>
+            <EuiComboBox
```

Comment:
**[minor] The styles for invalid state differs**
Styles for invalid field:
<img width="281" alt="Screenshot 2021-05-27 at 11 24 42" src="https://user-images.githubusercontent.com/4283304/119801694-3e2bb100-bede-11eb-9246-c05a00c32247.png">
<img width="279" alt="Screenshot 2021-05-27 at 11 32 06" src="https://user-images.githubusercontent.com/4283304/119802816-2c96d900-bedf-11eb-8d0f-efd0da344787.png">

Passing a prop `isInvalid` with a condition as in 241 line to a comboBox adds the red bottom line.